src/Source/DoctrineDBALConnectionSourceIterator.php

```diff
@@ -80,13 +80,11 @@ public function rewind(): void
     {
         $statement = $this->connection->prepare($this->query);
 
-        // TODO: Keep only the if part when dropping support for Doctrine DBAL < 3.1
+        // TODO: Keep only the else part when dropping support for Doctrine DBAL < 3.1
         if (method_exists($statement, 'executeQuery')) {
             $this->result = $statement->executeQuery($this->parameters);
         } else {
-            $statement->execute($this->parameters);
-
-            $this->result = $statement;
+            $this->result = $statement->execute($this->parameters);
```

Comment:
Can you take a look @jordisala1991, I feel like the DBAL3 code was wrong in your fix https://github.com/sonata-project/exporter/pull/550/files

Reply:
I think the older version returns void or bool and you have to use the $statement (static analysis wont help here since they only run with highest deps)

Reply:
When I look at https://github.com/sonata-project/exporter/pull/540/files#diff-38640adca7b4122bbd27d6173df4371be37ef41c1e885c744844e67755b0d35dR82
To me, the older version was returning void and in 3.0 it returns a result.

Then https://github.com/sonata-project/exporter/pull/550/files was made to fix the 2.13 compatibility, so you added the if part with the executeQuery for the version 2. In the futur we need to keep the else part.

Reply:
Can you try to add a test that covers those lines?

Reply:
It require a `Doctrine\DBAL\Driver\Connection` and I dunno how to have one.

Reply:
I think it is provided by driver manager? https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/configuration.html

Reply:
Done ! And I find the right fix